src/Queries/TermsQuery.php

```diff
@@ -8,23 +8,29 @@ class TermsQuery implements Query
 
     protected array $value;
 
-    public static function create(string $field, array $value): static
+    public static function create(string $field, array $value, null | float $boost = null): static
     {
-        return new self($field, $value);
+        return new self($field, $value, $boost);
     }
 
-    public function __construct(string $field, array $value)
+    public function __construct(string $field, array $value, protected null | float $boost = null)
     {
         $this->field = $field;
         $this->value = $value;
     }
 
     public function toArray(): array
     {
-        return [
+        $terms = [
             'terms' => [
                 $this->field => $this->value,
             ],
         ];
+
+        if ($this->boost) {
```

Comment:
what if we want to apply boost of 0? this won't work as expected

Reply:
Great suggestion. I thought I'd keep it in line with the existing `MatchQuery` but this is better. I've updated the `MatchQuery` as well now, or would you like a separate PR for this?